extensions/wikia/IndexingPipeline/PipelineEventProducer.class.php

```diff
@@ -1,69 +1,173 @@
 <?php
 
 class PipelineEventProducer {
+	const WG_CONTENT_NAMESPACES_KEY = 'wgContentNamespaces';
 	const ARTICLE_MESSAGE_PREFIX = 'article';
+	const PRODUCER_NAME = 'MWEventsProducer';
+	const CONTENT = 'content';
+	const ACTION_CREATE = 'create';
+	const ACTION_UPDATE = 'update';
+	const ACTION_DELETE = 'delete';
 	/** @var PipelineConnectionBase */
 	protected static $pipe;
 
+	/**
+	 * @desc Send event to pipeline in old format.
+	 * @param $eventName
+	 * @param $pageId
+	 * @param array $params
+	 */
 	public static function send( $eventName, $pageId, $params = [ ] ) {
-		global $wgCityId;
-		$msg = new stdClass();
-		$msg->cityId = $wgCityId;
-		$msg->pageId = $pageId;
+		$msg = self::prepareMessage( $pageId );
 		$msg->args = new stdClass();
 		foreach ( $params as $param => $value ) {
 			$msg->args->{$param} = $value;
 		}
-		try {
-			self::getPipeline()->publish( implode( '.', [ self::ARTICLE_MESSAGE_PREFIX, $eventName ] ), $msg );
-		} catch ( Exception $e ) {
-			\Wikia\Logger\WikiaLogger::instance()->error( __METHOD__, [
-				'exception' => $e,
-				'event_name' => $eventName
-			] );
-		}
+		self::publish( implode( '.', [ self::ARTICLE_MESSAGE_PREFIX, $eventName ] ), $msg );
 	}
 
-	/** @return PipelineConnectionBase */
-	protected static function getPipeline() {
-		if ( !isset( self::$pipe ) ) {
-			self::$pipe = new PipelineConnectionBase();
+	/**
+	 * @desc Send event to pipeline in new format.
+	 * @param $action
+	 * @param $id
+	 * @param string $ns
+	 * @param array $data
+	 * @param array $flags
+	 */
+	public static function nSend( $action, $id, $ns = self::CONTENT, $data = [ ], $flags = [ ] ) {
+		$ns = strtolower($ns);
+		$route = implode( '.', array_merge( [ self::PRODUCER_NAME, "_action:" . $action,  "_namespace:" . $ns ], $flags,
+				// adds info about the message content
+				array_map( function ( $item ) {
+					return "_content:{$item}";
+				}, array_keys( $data ) ) )
```

Comment:
I have a few remarks here:
- Route creation looks complex. Where are the tests for it?
- Please consider extracting valid keys ( _action, _namespace, _content ) to make sure those are defined at the beginning of the class.
- Content flags creation looks like it could be a separate method too
- maybe it is worth to pass flag-generating params in a dictionary object instead of function params?
- the description states 'new format' but does not link to a description nor describes it. Plus, 'new' tends to age; maybe sendKeyed or sendWithFlags would be more verbose